Rebase

przemyslawfierek-silvair opened the pull request.

michallowasrzechonek-silvair committed:
mesh: Add mesh-io for Silvair Radio over UART or TCP

michallowasrzechonek-silvair committed:
mesh: Check that incoming packet fits in net key cache

michallowasrzechonek-silvair committed:
mesh: Replace aes_cmac with aes_cmac_one

michallowasrzechonek-silvair committed:
mesh: Add option to use openssl for crypto

jakubwitowski-silvair committed:
mesh: Add DBus method for updating sequence number

michallowasrzechonek-silvair committed:
mesh: Emit PropertiesChanged for IvIndex, IvUpdate and IvState

michallowasrzechonek-silvair committed:
mesh: Add IvState property

przemyslawfierek-silvair committed:
mesh: Add D-Bus API to 'Access Control List' management to the 'mesh-io-tcpserver'

This patch provides D-Bus API to access control management. From now it
is possible to grant or revoke access for 'uGW' using single D-Bus call.

przemyslawfierek-silvair committed:
mesh: Add D-Bus initialization API to 'mesh-io'

This patch provides API to perform D-Bus initialization to 'mesh-io'.
This should allow to create some D-Bus objects related to 'mesh-io'.

michallowasrzechonek-silvair committed:
mesh: Compatilibty fix for openssl-1.0.2

michallowasrzechonek-silvair committed:
mesh: Fix memory leaks in openssl crypto

przemyslawfierek-silvair committed:
mesh: Add documentation for AttachFD

michallowasrzechonek-silvair committed:
Revert "mesh: Remove redundant call to save node configuration"

This reverts commit a51871bd73cfd58e71d5eb179aad06528ed7256a.

michallowasrzechonek-silvair committed:
mesh: Implement AttachFD method

michallowasrzechonek-silvair committed:
mesh: Trigger deferred Attach/Leave after JoinComplete reply

In fd3532354, upstream introduced a timer that defers Attach and Leave
calls made from JoinComplete handlers by 1 second.

This makes everything slower, particularly our tests. Let's fix it by
triggering the timeout (almost) immediately after receiving JoinComplete
reply.

przemyslawfierek-silvair committed:
mesh: Add AttachUnix method

michallowasrzechonek-silvair committed:
mesh: Cap sequence overcommit

If application exchanges a large number of messages with a local node
(think: config server, during node initialization), the overcommit
mechanism might save *way* too large sequence number in the storage.

This will cause the node to exhaust sequence numbers if storage fails
between node initialization and a graceful restart. Let's cap the
overcommit, so that this doesn't happen.

przemyslawfierek-silvair committed:
mesh: Add radio statistic message to the silvair_io

michallowasrzechonek-silvair committed:
mesh: Increase maximum number of subscriptions

przemyslawfierek-silvair committed:
mesh: Increase Silvair IO buffer for the incoming data.

This change increases IO buffer for the incoming data. This allows to
reduce number of IO operation and increase performance.

przemyslawfierek-silvair committed:
mesh: Add token bucket to limit number of processed packets.

Due to lack of resources on the Rigado Cascade 500 the number
of processed has to be limited to avoid the device kill.
This change allow to limit number of processed packets.

michallowasrzechonek-silvair committed:
mesh: Remove limits on model bindings

michallowasrzechonek-silvair committed:
mesh: Remove limits on model subscriptions

michallowasrzechonek-silvair commented:
Let's squash the final amqp commits to make history a bit less noisy.

There are also a few fixes to openssl crypto, so maybe squash them as well?

przemyslawfierek-silvair committed:
mesh: Add support to forwarding Mesh messages to AMQP broker

This patch adds AMQP client which allows to forward Mesh traffic to the
AMQP broker. BlueZ allows to subscribe topics with messages in FD
message format. All messages published on those topics will be send to
the Mesh network. Each Mesh message received by BlueZ will be published
to the set `exchange` with routing key `mon.<set_identity>.raw`. AMQP
functionality is active when `identity`, `exchange` and `url` is set.
